src/commands/commit/index.js

```diff
@@ -3,13 +3,23 @@ import inquirer from 'inquirer'
 
 import getEmojis from '../../utils/getEmojis'
 import prompts from './prompts'
-import withHook, { registerHookInterruptionHandler } from './withHook'
+import withHook, {
+  registerHookInterruptionHandler,
+  cancelIfRebasing
+} from './withHook'
 import withClient from './withClient'
 
 const commit = (mode: 'client' | 'hook') => {
-  if (mode === 'hook') registerHookInterruptionHandler()
+  if (mode === 'hook') {
+    registerHookInterruptionHandler()
+    return cancelIfRebasing().then(() => promptAndCommit(mode))
+  }
 
-  return getEmojis()
+  return promptAndCommit(mode)
+}
+
+const promptAndCommit = (mode: 'client' | 'hook') =>
```

Comment:
Can you create a `type CommitMode` for the `client` | `hook` thing ? Then we wont have to repeat those values!